src/gui/trackersadditiondialog.cpp

```diff
@@ -74,7 +76,13 @@ TrackersAdditionDialog::~TrackersAdditionDialog()
 
 void TrackersAdditionDialog::onAccepted() const
 {
-    const QVector<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(m_ui->textEditTrackersList->toPlainText());
+    const QVector<BitTorrent::TrackerEntryStatus> currentTrackers = m_torrent->trackers();
+    const int baseTier = !currentTrackers.isEmpty() ? (currentTrackers.last().tier + 1) : 0;
+
+    QVector<BitTorrent::TrackerEntry> entries = BitTorrent::parseTrackerEntries(m_ui->textEditTrackersList->toPlainText());
+    for (BitTorrent::TrackerEntry &entry : entries)
+        entry.tier = Utils::Number::add(entry.tier, baseTier);
```

Comment:
Wouldn't it make sense to add `baseTier` parameter to `
` function?

Reply:
>Wouldn't it make sense to add baseTier parameter to  function?

I guess you meant `parseTrackerEntries` function?
It wouldn't fit the case in sessionimpl.cpp where `m_additionalTrackerEntries` is built before appended to the target list.